Tasks/ServiceFabricUpdateManifestsV2/Update-DockerImageSettings.psm1

```diff
@@ -12,18 +12,44 @@ function Update-DockerImageSettings
         $appPackagePathSearchPattern = (Get-VstsInput -Name applicationPackagePath -Require)
         $appPackagePath = Find-VstsFiles -LegacyPattern $appPackagePathSearchPattern -IncludeDirectories
 
-        # Collect the image digest values for the images
+        $imageNamesFilePathSearchPattern = Get-VstsInput -Name imageNamesPath
+        $imageNamesPath = $null
+        if (-not [string]::IsNullOrEmpty($imageNamesFilePathSearchPattern))
+        {
+            $imageNamesPath = Find-VstsFiles -LegacyPattern $imageNamesFilePathSearchPattern
+        }
+
         $imageDigestValues = (Get-Content -LiteralPath $imageDigestsPath).Replace("`r`n", "`n").Split("`n")
+        $imageNames = $null
+        if (($imageNamesPath -ne $null))
+        {
+            $imageNames = (Get-Content -LiteralPath $imageNamesPath).Replace("`r`n", "`n").Split("`n")
+            if ($imageNames.Count -ne $imageDigestValues.Count)
```

Comment:
There seems to be an additional dependency which can be problematic:

1. Customers need to understand this additional input. And this will need to be mandatory whenever source image name contains a tag.
2. It assumes that image digest file and image name files contain entries in same order. Any change in this behavior in docker task(which is an independent task) will break the logic here.

Can we not take dependency on this images name file? We can update image digest file to contain both source and target images. 
Or may be just ignore tags part of image when updating manifest. [This can be problematic when two containers refer to same image but with different tags. Is this a practical scenario?]

Reply:
Updating the digest file would be a breaking change in the NuGet, the tools and the task which seems like a non-starter to me.

It is unlikely that anyone in a real scenario would ever deploy an application with two images of the same name and different tags. But I am hesitant to completely block that scenario. We would likely need to add an error check to block b/c otherwise it would not get noticed until post deployment. @bishal-pdMSFT: Would you rather eliminate that scenario than add the new input?

If it helps as a part of this change I was also going to update the template to fill in the correct value so that future build definitions would just work.

Reply:
I am not able to foresee a scenario where different tags will be required. If I understand correctly build tools is generating this tag - user only specifies a dockerfile. Why will tools decide to assign different tags or for that matter why will it build different images for same dockerfile. IMO, unless we can see a strong possibility of this, we should try to keep task simple and not add a new input.
+1 for template change if required.

Reply:
The build tools do not generate the tag. There is a tag provided in the templates, but it is up to the user.